gcsio/src/main/java/com/google/cloud/hadoop/gcsio/StorageStubProvider.java

```diff
@@ -145,12 +142,14 @@ public int activeRequests() {
   StorageStubProvider(
       GoogleCloudStorageOptions options,
       ExecutorService backgroundTasksThreadPool,
-      Credential credential) {
+      Credentials credentials,
+      boolean useDirectpath) {
     this.readOptions = options.getReadChannelOptions();
     this.userAgent = options.getAppName();
     this.backgroundTasksThreadPool = backgroundTasksThreadPool;
     this.mediaChannelPool = new ArrayList<>();
-    this.credential = checkNotNull(credential, "credential can not be null");
+    this.credentials = credentials;
+    this.useDirectpath = useDirectpath;
```

Comment:
Instead of using flags and branching logic, can we override the relevant methods in a child class ?

ex. An `abstract` class with common logic, override `newBlockingStub` and `newAsycStub` in child classes

Reply:
Done.